src/python/pants/backend/python/subsystems/setup.py

```diff
@@ -138,12 +138,36 @@ def register_options(cls, register):
             default={"python-default": "3rdparty/python/default_lock.txt"},
             help=(
                 "A mapping of logical names to lockfile paths used in your project.\n\n"
-                "For now, things only work properly if you define a single resolve and set "
-                "`[python].experimental_default_resolve` to that value. We are close to "
-                "properly supporting multiple (disjoint) resolves.\n\n"
-                "To generate a lockfile, run `./pants generate-lockfiles --resolve=<name>` or "
-                "`./pants generate-lockfiles` to generate for all resolves (including tool "
-                "lockfiles).\n\n"
+                "Many organizations only need a single resolve for their whole project, which is "
+                "a good default and the simplest thing to do. However, you may need multiple "
+                "resolves, such as if you use two conflicting versions of a requirement in "
+                "your repository.\n\n"
+                "For now, Pants only has first-class support for disjoint resolves, meaning that "
+                "you cannot ergonomically set a `python_source` target, for example, to work "
+                "with multiple resolve. Practically, this means that you cannot yet reuse common "
+                "code, such as util files, across projects using different resolves. Support for "
+                "overlapping resolves is coming soon.\n\n"
+                "If you only need a single resolve, optionally override this option's default if "
+                "you would like to write your project's lockfile to a different path. Make "
+                "sure that `[python].experimental_default_resolve` is set to the same resolve "
+                "name. Then, run `./pants generate-lockfiles --resolve=<name>` to generate the "
+                "lockfile.\n\n"
```

Comment:
```suggestion
                "To generate a lockfile, run `./pants generate-lockfiles --resolve=<name>`.\n\n"
```

IMO, you can drop and/or de-emphasize the "what if I need to rename the lockfile" case for the single-resolve case. Since this will be a new lockfile format, we should not expect that anyone already has one, and so not many folks should need to override it.